tests/test-team/template-mgmt-component-tests/template-mgmt-create-nhs-app-template-page.component.ts

```diff
@@ -32,15 +50,193 @@ test.describe('Create NHS App Template Page', () => {
+      'Create NHS App message template'
+    );
+    const templateName = 'NHS Testing 123';
+    await page.locator('[id="nhsAppTemplateName"]').fill(templateName);
+    await createTemplatePage.clickContinueButton();
+    await expect(page.locator('.nhsuk-error-summary')).toBeVisible();
+    await expect(
+      page.locator('ul[class="nhsuk-list nhsuk-error-summary__list"] > li')
+    ).toHaveText(['Enter a template message']);
+  });
+
+  test('Validate error messages on the create NHS App message template page with no template name', async ({
+    page,
+    baseURL,
+  }) => {
+    const createTemplatePage = new TemplateMgmtCreatePage(page);
+
+    await createTemplatePage.loadPage(emptySessionData.id);
 
     await expect(page).toHaveURL(
-      `${baseURL}/templates/create-nhs-app-template/${nhsAppNoTemplateSessionData.id}`
+      `${baseURL}/templates/create-nhs-app-template/${emptySessionData.id}`
     );
     expect(await createTemplatePage.pageHeader.textContent()).toBe(
       'Create NHS App message template'
     );
+    const templateMessage = 'Test Message box';
+    await page.locator('[id="nhsAppTemplateMessage"]').fill(templateMessage);
+    await createTemplatePage.clickContinueButton();
+    await expect(page.locator('.nhsuk-error-summary')).toBeVisible();
+
+    await expect(
+      page.locator('ul[class="nhsuk-list nhsuk-error-summary__list"] > li')
+    ).toHaveText(['Enter a template name']);
+  });
+
+  test('5000 words Entered in Template body and word count correctly displayed', async ({
+    page,
+    baseURL,
+  }) => {
+    const createTemplatePage = new TemplateMgmtCreatePage(page);
+
+    await createTemplatePage.loadPage(emptySessionForTemplateCreation.id);
+
+    await expect(page).toHaveURL(
+      `${baseURL}/templates/create-nhs-app-template/${emptySessionForTemplateCreation.id}`
+    );
+    const templateName = 'NHS Testing 123';
+    await page.locator('[id="nhsAppTemplateName"]').fill(templateName);
+    const templateMessage = 'T'.repeat(5000).trim();
+    await page.locator('[id="nhsAppTemplateMessage"]').fill(templateMessage);
+    await expect(page.getByText('5000 of 5000 characters')).toBeVisible();
+  });
+
+  test('5001 words attempted to be entered in Template body and only 5000 allowed', async ({
+    page,
+    baseURL,
+  }) => {
+    const createTemplatePage = new TemplateMgmtCreatePage(page);
+
+    await createTemplatePage.loadPage(emptySessionForTemplateCreation.id);
+
+    await expect(page).toHaveURL(
+      `${baseURL}/templates/create-nhs-app-template/${emptySessionForTemplateCreation.id}`
+    );
+    const templateName = 'NHS Testing 123';
+    await page.locator('[id="nhsAppTemplateName"]').fill(templateName);
+    const templateMessage = 'T'.repeat(5001).trim();
+    await page.locator('[id="nhsAppTemplateMessage"]').fill(templateMessage);
+    await expect(page.getByText('5000 of 5000 characters')).toBeVisible();
```

Comment:
Is the expected behaviour or the page to not accept more than 5000 characters or to stop counting after 5000 characters?

If it stops accepting more than 5000 characters might be worth doing an additional assert that checks that only 5000 characters are present

If it stops counting after 5000 characters but still allows you to submit more, it might be worth checking if theres a validation error message that prevents the user from going any further in their journey?

Reply:
I'll double check this one but believe it should stop accepting text in the input field so the additional assertion is a good shout.